mmv1/third_party/terraform/services/resourcemanager/resource_google_project_iam_member_remove.go

```diff
@@ -0,0 +1,109 @@
+package resourcemanager
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgiamresource"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func ResourceGoogleProjectIamMemberRemove() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoogleProjectIamMemberRemoveCreate,
+		Read:   resourceGoogleProjectIamMemberRemoveRead,
+		Delete: resourceGoogleProjectIamMemberRemoveDelete,
+
+		CustomizeDiff: customdiff.All(
+			tpgresource.DefaultProviderProject,
+		),
+
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The project that the service account will be created in. Defaults to the provider project configuration.`,
+			},
+			"member": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `The Identity of the service account in the form 'serviceAccount:{email}'. This value is often used to refer to the service account in order to grant IAM permissions.`,
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceGoogleProjectIamMemberRemoveCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	project := d.Get("project").(string)
+	role := d.Get("role").(string)
+	member := d.Get("member").(string)
+
+	iamPolicy, err := config.NewResourceManagerClient(config.UserAgent).Projects.GetIamPolicy(project,
+		&cloudresourcemanager.GetIamPolicyRequest{
+			Options: &cloudresourcemanager.GetPolicyOptions{
+				RequestedPolicyVersion: tpgiamresource.IamPolicyVersion,
+			},
+		}).Do()
+	for _, bind := range iamPolicy.Bindings {
+		for _, existingMember := range bind.Members {
+			if member == existingMember {
+				if role == bind.Role {
```

Comment:
Can you test this with multiple roles assigned to the identity we're removing the single-role membership from?

Reply:
Good catch, looks like for each binding, there's one role matching to one member. I think this might cover it already but I need to verify.
https://stackoverflow.com/questions/61661116/want-to-assign-multiple-google-cloud-iam-roles-to-a-service-account-via-terrafor

Reply:
added a test for multiple members on a same role, test says passed, but it's not behaving correctly, will attempt to fix that.